### PR TITLE
More Arc to `AggregateFunctionExpr`

### PR DESCRIPTION
**Pull request by alamb**
Targets https://github.com/apache/datafusion/pull/12950

This is a proposal to add even more `Arc` to the `AggregateFunctionExpr` in order to reduce more deep clones

### CODE REVIEW

File: datafusion/physical-expr/src/aggregate.rs [hunk 1]
```diff
@@ -331,17 +331,15 @@ impl AggregateFunctionExpr {
...
-            .with_beneficial_ordering(beneficial_ordering)?
+        let Some(updated_fn) =
+            Arc::clone(&self.fun).with_beneficial_ordering(beneficial_ordering)?
         else {
             return Ok(None);
         };
 
         AggregateExprBuilder::new(Arc::new(updated_fn), self.args.to_vec())
             .order_by(self.ordering_req.to_vec())
-            .schema(Arc::new(self.schema.clone()))
```
Comment (alamb) [inline datafusion/physical-expr/src/aggregate.rs:344]: In particular, I believe this is a deep clone

File: datafusion/physical-expr/src/aggregate.rs [hunk 2]
```diff
@@ -331,17 +331,15 @@ impl AggregateFunctionExpr {
         self: Arc<Self>,
         beneficial_ordering: bool,
     ) -> Result<Option<AggregateFunctionExpr>> {
-        let Some(updated_fn) = self
-            .fun
-            .clone()
```
Comment (alamb) [inline datafusion/physical-expr/src/aggregate.rs:336]: this was also a deep clone

**Comment by alamb** [comment]
Will create a PR to DataFusion main instead

**Comment by askalt** [comment]
> Will create a PR to DataFusion main instead

Ok, thank you, I was busy yesterday, I didn't have time to merge it :(.

**Pull request closed without merging**